src/conn/conn_api.c

```diff
@@ -2974,7 +2979,34 @@ wiredtiger_open(const char *home, WT_EVENT_HANDLER *event_handler, const char *c
      */
     WT_ERR(__wt_config_gets(session, cfg, "verify_metadata", &cval));
     verify_meta = cval.val;
-    WT_ERR(__wt_turtle_init(session, verify_meta));
+    WT_ERR(__wt_config_gets(session, cfg, "backup_partial_restore", &cval));
```

Comment:
This code (or the new hash code it becomes) should be in its own function. You can send in the `cfg` and check for the list in the function.